File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int[] arr = new int[in.nextInt()];
        int tmp = 0, num = in.nextInt();
        for(int i = 0; i < arr.length; i++)
            arr[i] = i+1;
        for(int i = 1; i <= num; i++){
            int a = in.nextInt()-1, b = in.nextInt()-1;
            for(int j = 0; j <= (b-a-1)/2; j++){
                tmp = arr[a+j];
                arr[a+j] = arr[b-j];
                arr[b-j] = tmp;
            }
        }
        for(int x:arr)  System.out.printf("%d ", x);
    }
}

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.kt
import java.util.Scanner
fun main(){
    val sc = Scanner(System.`in`)
    var arr = IntArray(sc.nextInt(),{0})
    for(i in 0..(arr.size-1)){
        arr[i] = i+1
    }
    for(i in 1..sc.nextInt()){
        var a = sc.nextInt()-1; var b = sc.nextInt()-1
        for(j in 0..(b-a-1)/2){
            var tmp = arr[a+j]
            arr[a+j] = arr[b-j]
            arr[b-j] = tmp
        }
    }
    arr.forEach { print("$it ") }
}